go-controller/pkg/ovn/namespace.go

```diff
@@ -244,12 +244,5 @@ func (oc *Controller) getNamespaceLock(ns string) *sync.Mutex {
 
 	// lock the individual namespace
 	mutex.Lock()
-
-	// check that the namespace wasn't deleted between getting the two locks
```

Comment:
You can't remove this code. Note that we unlock namespaceMutexMutex before we grab this lock, so we could lose the race and have the namespace be deleted before we lock it.

Reply:
@squeed we can't access oc.namespaceMutex without holding a lock though. mutex.Lock() doesn't count, because that doesn't protect against concurrent access when we start parallelizing multiple pods. It only works now because we serialize pod events, and that's totally non-obvious.

I guess we can keep it for now (the real bug I was fixing is the updateNamespace() locking) and fix it with address set objects that have their own lock.

Reply:
Oh crap, you're right - that `namespaceMutex` call on line 249 is a race.

The goal is to be able to wait for the namespace mutex without blocking others waiting on namespaceMutexMutex. So I guess we need to lock namespaceMutexMutex again and do the check.